src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-details/rgw-user-details.component.spec.ts

```diff
@@ -66,4 +68,48 @@ describe('RgwUserDetailsComponent', () => {
     expect(detailsTab[14].textContent).toEqual('MFAs(Id)');
     expect(detailsTab[15].textContent).toEqual('testMFA1, testMFA2');
   });
+  it('should test updateKeysSelection', () => {
+    component.selection = {
+      hasMultiSelection: false,
+      hasSelection: false,
+      hasSingleSelection: false,
+      _selected: []
+    };
+    component.updateKeysSelection(component.selection);
+    expect(component.keysSelection).toEqual(component.selection);
+  });
+  it('should call showKeyModal when key selection is of type S3', () => {
+    component.keysSelection.first = () => {
+      return { type: 'S3', ref: { user: '', access_key: '', secret_key: '' } };
+    };
+    const s = spyOn(component['modalService'], 'show').and.callFake(() => {
```

Comment:
a better name would be nice for people who look at it later.
```suggestion
    const modalShowSpy = spyOn(component['modalService'], 'show').and.callFake(() => {
```

Reply:
Updated